crates/ty_python_semantic/resources/mdtest/narrow/isinstance.md

```diff
@@ -289,6 +288,23 @@ def _(flag: bool):
         reveal_type(x)  # revealed: Literal[1, "a"]
 ```
 
+## Generic aliases are not supported as second argument
+
+The `classinfo` argument cannot be a generic alias:
+
+```py
+def _(x: list[str] | list[int] | list[bytes]):
+    # TODO: Ideally, this would be an error (requires https://github.com/astral-sh/ty/issues/116)
+    if isinstance(x, list[int]):
+        # No narrowing here:
+        reveal_type(x)  # revealed: list[str] | list[int] | list[bytes]
+
+    # error: [invalid-argument-type] "Invalid second argument to `isinstance`"
+    if isinstance(x, list[int] | list[str]):
+        # No narrowing here:
+        reveal_type(x)  # revealed: list[str] | list[int] | list[bytes]
```

Comment:
could we also add an examples like:

```py
ListStrOrInt = Union[list[str], int]

def _(x: dict[int, str] | ListStrOrInt):
    if isinstance(x, ListStrOrInt):
        reveal_type(x)

    if isinstance(x, Union[list[str], int]):
        reveal_type(x)
```

Both fail at runtime, so ideally we wouldn't do any narrowing (or we'd intersect with `Unknown` rather than `list[str] | int`), and ideally we'd flag the error with a diagnostic.

But this very rarely comes up, so it's obviously fine if this fails right now and we just add TODOs to the test